File: WinForms/CS/MVVMMainDemo/Modules/TodoApp/TodoAppModule.cs
using System.Windows.Forms;

namespace DevExpress.MVVM.Demos.TodoApp {
    public class TodoAppModule : TutorialControl {
        Form todoAppForm;
        public TodoAppModule() {
            todoAppForm = new global::TodoApp.MainView();
            Tutorials.TodoApp.PresentationHelper.CenterParent(this, todoAppForm, true);
        }
        protected override void DoShow() {
            base.DoShow();
            if(todoAppForm != null && !todoAppForm.Visible)
                todoAppForm.Show();
        }
        protected override void DoHide() {
            if(todoAppForm != null && todoAppForm.Visible)
                todoAppForm.Hide();
            base.DoHide();
        }
    }
}

File: WinForms/VB/MVVMMainDemo/Modules/TodoApp/TodoAppModule.vb
Namespace DevExpress.MVVM.Demos.TodoApp
    Public Class TodoAppModule
        Inherits TutorialControl
        Private todoAppForm As Form
        Public Sub New()
            todoAppForm = New Global.TodoApp.MainView()
            Tutorials.TodoApp.PresentationHelper.CenterParent(Me, todoAppForm, True)
        End Sub
        Protected Overrides Sub DoShow()
            MyBase.DoShow()
            If todoAppForm IsNot Nothing AndAlso Not todoAppForm.Visible Then todoAppForm.Show()
        End Sub
        Protected Overrides Sub DoHide()
            If todoAppForm IsNot Nothing AndAlso todoAppForm.Visible Then todoAppForm.Hide()
            MyBase.DoHide()
        End Sub
    End Class
End Namespace
